crates/hir_ty/src/lib.rs

```diff
@@ -1053,13 +1064,15 @@ mod tests {
         check_in_root_file(
             r#"
                 fn main() {
-                    let a = true
+                    let a = true;
+                    let mut b = 10;
                 }
             "#,
             expect![[r#"
                 //- /main.nail
                 fn entry:main() -> () {
                     let a = true; //: bool
+                    let mut b = 10; //: int
```

Comment:
**Optimization Issue**: Introduction of mutable variable 'b' which is not used elsewhere in the provided context. This could lead to unnecessary memory allocation and management overhead. <br> **Fix**: If the variable 'b' is not used meaningfully in subsequent code, consider removing it or ensure that it is utilized effectively to justify its mutability and existence. <br> **Code Suggestion**: 
 ```
 -                    let mut b = 10;
 -                    let mut b = 10; //: int
 ```